pkg/target/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,335 @@
+				lastErr = err
+				if i == p.retryConfig.MaxRetries {
+					break
+				}
+
+				zap.L().Warn("PagerDuty API call failed, retrying",
+					zap.Int("attempt", i+1),
+					zap.Int("maxRetries", p.retryConfig.MaxRetries),
+					zap.Duration("wait", wait),
+					zap.Duration("maxWait", p.retryConfig.MaxWait),
+					zap.String("target", p.Name()),
+					zap.Error(err),
+				)
+
+				timer := time.NewTimer(wait)
+				select {
+				case <-ctx.Done():
+					timer.Stop()
+					return fmt.Errorf("operation timed out during retry: %w", ctx.Err())
+				case <-timer.C:
+				}
+
+				wait *= 2
+				if wait > p.retryConfig.MaxWait {
+					wait = p.retryConfig.MaxWait
+				}
+				continue
+			}
+			return nil
+		}
+	}
+
+	return fmt.Errorf("operation failed after %d retries: %w", p.retryConfig.MaxRetries, lastErr)
+}
+
+// findExistingIncident searches for an open incident matching the given result
+// Returns the matching incident if found, nil otherwise
+func (p *client) findExistingIncident(ctx context.Context, result v1alpha2.PolicyReportResult) (*pagerduty.Incident, error) {
+	// Query for open incidents with matching metadata
+	opts := pagerduty.ListIncidentsOptions{
+		ServiceIDs: []string{p.serviceID},
+		Statuses:   []string{"triggered", "acknowledged"},
+	}
+	
+	incidents, err := p.client.ListIncidents(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list incidents: %w", err)
+	}
+
+	metadata := incidentMetadata{
+		ClusterName: p.clusterName,
+		PolicyName:  result.Policy,
+		RuleName:    result.Rule,
+	}
+	if result.HasResource() {
+		res := result.GetResource()
+		metadata.ResourceKind = res.Kind
+		metadata.ResourceNS = res.Namespace 
+		metadata.ResourceName = res.Name
+	}
+
+	// Find matching incident by checking custom fields
+	for _, incident := range incidents {
+		if matchesMetadata(incident, metadata) {
+			return &incident, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// resolveIncident marks an incident as resolved with the given resolution message
+func (p *client) resolveIncident(ctx context.Context, incidentID string, resolution string) error {
+	incident := pagerduty.ManageIncidentsOptions{
+		ID: incidentID,
+		Incidents: []pagerduty.ManageIncident{
+			{
+				Status:     "resolved",
+				Resolution: resolution,
+			},
+		},
+	}
+
+	return p.retryWithExponentialBackoff(ctx, func() error {
+		return p.client.ManageIncidents("policy-reporter", &incident)
+	})
+}
+
+// Send processes a policy result and manages corresponding PagerDuty incidents
+// - Creates new incidents for failing policy results
+// - Resolves incidents when policies pass or are deleted
+// - Deduplicates incidents for the same policy violation
+func (p *client) Send(result v1alpha2.PolicyReportResult) {
+	ctx, cancel := context.WithTimeout(context.Background(), p.timeoutConfig.Create)
+	defer cancel()
+
+	// For pass results or deletions, resolve any existing incidents
+	if result.Result == v1alpha2.StatusPass || result.Result == "" {
```

Comment:
Targets only receive fail / warn / error results in the Send method.

Currently the SecurityHub is the only target that has the possibility to resolve fails after a result changes or is removed. Maybe its worth looking into it and check how we can handle it for this target as well.